code-challenges.js

```diff
@@ -51,18 +96,82 @@ var stringWithVowels2 = "ILoveJavaScript"
+  let myFilteredArray = stringSplitter.filter(value => {
+    return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u" && value !== "A" && value !== "E" && value !== "I" && value !== "O" && value !== "U"
+  })
+  return myFilteredArray.join("")
+} else {
+  return `"${array}" is not a string!`
+  }
+}
+
+console.log(isItAString(notAString1))
+console.log(isItAString(notAString2))
+
+
+// this is all the work i originally did on this question & I kept getting more and more confused!!
+
+//  i thought about adding an if statement at the beginning , or the other option was to refactor it and use a for loop to run through the array and
+// const removeVowels = (array) => if (array !==) {
+//
+// } else if{
+//
+// }{
+//   return array.filter(value => {
+//     return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u" && value !== "A" && value !== "E" && value !== "I" && value !== "O" && value !== "U"
+//   })
+// }
+
+
+// const removeVowels2 = (array) => {
+//   let newArray = []
+//   for (let i=0; i<array.length; i++){
+//     if (typeof array[i] !== "string"){
+//       return `${array} is not a string`
+//     } else if (typeof array[i]) === "string" {
+//       return array.filter(value => {
+//         return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u" && value !== "A" && value !== "E" && value !== "I" && value !== "O" && value !== "U"
+//     }
+//
+// console.log(removeVowels2(notAString1))
+
+// const removeVowels2 = (string) => {
+//   var stringCheck === typeof string
+//   if (stringCheck !== "string" ){
+//     return `${string} is not a string`
+//   } else {
+//
+//   }
+// }
+
 
 
 
 
 // --------------------6) Create a function that takes the toonimals array and returns an array with only the toon objects that are cats.
 // Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
 
-var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
+var toonimals = [
+  { name: "Itchy", animal: "mouse" },
+  { name: "Stimpy", animal: "cat" },
+  { name: "Daffy", animal: "duck" },
+  { name: "Scratchy", animal: "cat" },
+  { name: "Ren", animal: "dog" },
+  { name: "Felix", animal: "cat" }]
 
+//write a function that takes an array as an argument and filters it out to return only the toon objects that have thei "value" in the  set to cats
 
+var catChecker = toonimals.filter(value => value.animal === "cat")
+console.log(catChecker)
 
 
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+
+var animalChecker = toonimals.filter(value => value.animal !== "cat")
+console.log(animalChecker)
```

Comment:
same here! also, we only want to return the _name_ of the cat, not the whole object ;)